### PR TITLE
[no ticket][risk=no] restore auth debugging line

### PR DESCRIPTION
**Pull request by jaycarlton**
Restore an apparently useful log statement.
---
**PR checklist**

- [ ] This PR meets the Acceptance Criteria in the JIRA story
- [ ] The JIRA story has been moved to Dev Review
- [ ] This PR includes appropriate unit tests
- [ ] I have run and tested this change locally
- [ ] If this includes an API change, I have updated the appropriate Swagger definitions and notified API consumers
- [ ] If this includes a new feature flag, I have created and linked new JIRA tickets to (a) turn on the feature flag and (b) remove it later

### CODE REVIEW

**Comment by jaycarlton** [comment]
@calbach fyi

File: api/src/main/java/org/pmiops/workbench/interceptors/AuthInterceptor.java [hunk 1]
```diff
@@ -171,7 +171,7 @@ public boolean preHandle(HttpServletRequest request, HttpServletResponse respons
             new UserAuthentication(user, OAuth2Userinfo, token, UserType.RESEARCHER));
 
     // TODO: setup this in the context, get rid of log statement
```
Comment (gjuggler) [inline api/src/main/java/org/pmiops/workbench/interceptors/AuthInterceptor.java:173]: Might be worth converting this vague (and misleading?) TODO into an explainer comment describing why this log line is important. Channeling CH, it sounds like this is important to verify which user an authenticated request came from.

Comment (calbach) [inline api/src/main/java/org/pmiops/workbench/interceptors/AuthInterceptor.java:173]:
```suggestion
    // TODO: Remove log statement and attach userinfo as a Stackdriver key/value instead.
    // This log line is currently the the only reliable way to associate a particular App Engine request log
    // with the authenticated user identity, which is critical information for debugging.
```

Comment (jaycarlton) [reply api/src/main/java/org/pmiops/workbench/interceptors/AuthInterceptor.java:173]: @calbach if you can sketch out which properties you want in a Stackdriver log (and if you want it in JSON, which I recommend for filtering and making logs-based metrics), what keys & values would be helpful, I'll be happy to do the logging part.

Comment (jaycarlton) [reply api/src/main/java/org/pmiops/workbench/interceptors/AuthInterceptor.java:173]: merging this in the meantime

Comment (calbach) [reply api/src/main/java/org/pmiops/workbench/interceptors/AuthInterceptor.java]: Just email is fine to start to keep parity here. I don't have other ideas off the top of my head (authority and data access level are also interesting)

**Pull request merged**